lib/exq/manager/server.ex

```diff
@@ -251,6 +281,17 @@ defmodule Exq.Manager.Server do
 
       job_results = jobs |> Enum.map(fn(potential_job) -> dispatch_job(state, potential_job) end)
 
+      # Update worker info in redis that it is alive
```

Comment:
So this manages worker state each time it dequeues, is this how Sidekiq does it?

Reply:
This seems to be adding overhead.  Looking at performance test:

without lines 284:293
`2017-10-29 13:25:34.217 [debug]: Perf test took 0.680126 secs`

with lines 284:293
`2017-10-29 13:23:03.301 [debug]: Perf test took 1.000115 secs`

(To run this I set `ExUnit.start(capture_log: false)` in test_helper.exs)

Reply:
Hi @akira, biggest problem is cleanup of queues which happens parallel by calling: https://github.com/akira/exq/pull/289/files/adbd0d7657abff63dcdfdb1db5a7240c7ce4d4a6#diff-ee19dd3b88a278d7cf31b33f2b2b5705R147

I am new to elixir so I have no experience how to do it in one run, first, cleanup, then whole code.

Reply:
then you wil need just: 
```
name = redis_worker_name(state)
      worker_init = [
         ["HSET", name, "beat", Time.unix_seconds],
         ["EXPIRE", name, (state.poll_timeout / 1000 + 5)], # expire information about live worker in poll_interval + 5s
       ]
Connection.qp!(state.redis, worker_init)
```

Reply:
also this code can run every second and result will be same

Reply:
Yes - for Sidekiq, seems like it is running this in a separate thread:
https://github.com/mperham/sidekiq/blob/master/lib/sidekiq/launcher.rb#L25
That is why I was thinking if we can just run this separately as well, would it be possible to do this in a separate GenServer?  

If you need to do synchronous methods from the Manager to the GenServer, you can still do GenServer.call instead of GenServer.cast and it will wait for the call. Let me know if you think this is doable or not.  The other option is perhaps lower the frequency of this call so it doesn't do it in every dequeue.   

The other thing I noticed is the section with HSET, etc is duplicated - it would be nice to take this out into a function at least, or function in JobStat module.